pom.xml

```diff
@@ -21,10 +21,10 @@
     <artifactId>swarm-plugin</artifactId>
     <packaging>pom</packaging>
     <name>Jenkins Self-Organizing Swarm Plug-in</name>
-    <version>3.20-SNAPSHOT</version>
+    <version>3.21-SNAPSHOT</version>
 
     <properties>
-        <jenkins.version>2.164.3</jenkins.version>
+        <jenkins.version>2.229</jenkins.version>
```

Comment:
Full WebSocket support via Remoting 4.3 is still not yet available in an LTS release, so I am hesitant to bump the minimum required Jenkins version to a non-LTS version. Does the Remoting client fail cleanly if used with `-webSocket` against a Jenkins master with an earlier version of Remoting that does not support the WebSocket protocol? If so, then I think the best solution would be to keep the minimum Jenkins version as-is to allow users of older versions to keep using this plugin without the WebSocket protocol.

Can you please add an integration test to `plugin/src/test/java/hudson/plugins/swarm/SwarmClientIntegrationTest.java` that exercises the new WebSocket support? Assuming that `jenkins.version` remains at 2.164.3, when running `mvn test` this would be expected to fail. Ideally we can get this to fail with a clean error message, but if that isn't feasible then documenting the requirement would be acceptable. The test should pass when running with a newer Jenkins version (e.g., by running `mvn test -Djenkins.version=2.229`). If you get this working, then you get the test suite to be clean on both Jenkins versions by adding a line like `Assume.assumeTrue(j.jenkins.getVersion().isNewerThan(new VersionNumber("2.229")));` to the top of the test method. Once you get `mvn test -Djenkins.version=2.229` working locally, ping me and I will add that version to the matrix tested by the CI for this plugin so that we can ensure there are no regressions in the future.

Reply:
I will make the attempt, just stumbling on getting `mvn test -Djenkins.version=2.229` to run
which is a problem on my side i will assume

Reply:
Thanks! I just tested this locally and it works for me using the following commands:

```
$ java -version                            
openjdk version "1.8.0_252"
OpenJDK Runtime Environment (build 1.8.0_252-8u252-b09-1ubuntu1-b09)
OpenJDK 64-Bit Server VM (build 25.252-b09, mixed mode)
$ mvn -version                             
Apache Maven 3.6.3
Maven home: /usr/share/maven
Java version: 1.8.0_252, vendor: Private Build, runtime: /usr/lib/jvm/java-8-openjdk-amd64/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.4.0-29-generic", arch: "amd64", family: "unix"
$ mvn clean package -Djenkins.version=2.229
[…]
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running InjectedTest
[INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 6.487 s - in InjectedTest
[INFO] Running hudson.plugins.swarm.PipelineJobTest
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 37.096 s - in hudson.plugins.swarm.PipelineJobTest
[INFO] Running hudson.plugins.swarm.SwarmClientIntegrationTest
[INFO] Tests run: 23, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 119.854 s - in hudson.plugins.swarm.SwarmClientIntegrationTest
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 32, Failures: 0, Errors: 0, Skipped: 0
[…]
$
```

Reply:
yes this is just me just figuring out java

Reply:
Hi @basil i added the test but there is something weird going on  with the windows build. Dont know whether that is something related or not. most likely.